src/allmydata/immutable/downloader/node.py

```diff
@@ -76,7 +78,7 @@ def __init__(self, verifycap, storage_broker, secret_holder,
         # .guessed_segment_size, .guessed_num_segments, and
         # .ciphertext_hash_tree (with a dummy, to let us guess which hashes
         # we'll need)
-        self._build_guessed_tables(DEFAULT_MAX_SEGMENT_SIZE)
+        self._build_guessed_tables(self.default_max_segment_size)
```

Comment:
Uhh this is so weird.  Guess a value that is trivially readable from the front of any share and derive a bunch of real values from it -- then replace all those values with the real values after doing the trivial read -- supposedly as an optimization for `Segmentation` -- which cannot possibly work correctly if you guessed wrong? :/ :/ :/ :/

I guess I get to read some more of `allmydata/immutable/downloader/` now...

Reply:
If it can't work if you guess wrong then this PR is no good :(

Reply:
But if you can get the segment value from trivial read... I should do that instead?

Reply:
It seems like incorrect guesses don't impair functionality.  Maybe they make performance slightly worse or something (we could try to measure but I don't particularly want to at the moment).

So I think this part is fine as-is.